src/useIsShake.ts

```diff
@@ -51,14 +54,41 @@ export function useIsShake() {
     setSubscription(null);
   };
 
+  const checkAccelerometerAvail = async () => {
+    const availability = await isAvailableAsync();
+    setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return availability;
+    });
+  };
+
+  const getAccelerometerPermiss = async () => {
+    const permissionResponse = await requestPermissionsAsync();
+    if (permissionResponse.granted) setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return permissionResponse.granted;
+    }); // following else if block for when permissionResponse.canAskAgain === false in order to direct end user to Settings app in order to enable permission to access Accelerometer
+    else if (!permissionResponse.canAskAgain) {
+      Linking.openSettings();
+    };
```

Comment:
If a request for permission to access the accelerometer is denied, and cannot be asked for again, the end user will be directed to Settings for their device pertaining to enabling accelerometer access for app.

Reply:
Screenshots, GIF or video?